apps/condo/domains/ticket/schema/TicketComment.js

```diff
@@ -72,14 +76,53 @@ const TicketComment = new GQLListSchema('TicketComment', {
             schemaDoc: 'Plain text content',
             type: Text,
         },
-
     },
-    plugins: [uuided(), versioned(), tracked(), softDeleted(), historical()],
     hooks: {
+        resolveInput: async ({ operation, listKey, context, resolvedData, existingItem }) => {
+            const user = get(context, ['req', 'user'])
+            const userType = get(user, 'type')
+            const commentType = get(resolvedData, 'type')
+
+            if (operation === 'create' && commentType === RESIDENT_COMMENT_TYPE) {
+                const ticketId = get(resolvedData, 'ticket')
+                const dv = get(resolvedData, 'dv')
+                const sender = get(resolvedData, 'sender')
+                const now = new Date().toISOString()
+
+                if (userType === RESIDENT) {
+                    await Ticket.update(context, ticketId, {
+                        dv,
+                        sender,
+                        lastResidentCommentAt: now,
+                    })
```

Comment:
What about moving this part of code outside of both branches of the condition?

Reply:
if userType is resident then we need to update `lastResidentCommentAt`
else if userType is not resident then we need to update `lastAnsweredToResidentAt`

Not sure if this needs to be moved outside of the if-else